js/weekView.js

```diff
@@ -127,27 +127,30 @@ const generateWeekView = (date) => {
 
 export const switchWeekView = (newDate) => {
```

Comment:
you pass in newDate as arg to this function, but then don't actually use it
It's generally better for functions to be "pure", to not implicitly depend on state, if you can avoid it.
Just pass in newDate, oldDate and use them explicitly.